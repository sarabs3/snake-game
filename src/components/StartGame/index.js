import React from 'react';

const StartGame = props => {
	return (
		<div className="snake-app__overlay">
			<button onClick={props.start} data-testid={'start-game'}>Start game!</button>
		</div>
	);
};

export default StartGame;
