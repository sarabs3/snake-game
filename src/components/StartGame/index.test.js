import React from 'react';
import ReactDOM from 'react-dom';

import renderer from 'react-test-renderer';
import {render, fireEvent} from '@testing-library/react';

import StartGame from './index';

let div;

beforeAll(() => {
	div = document.createElement('div');
	document.body.appendChild(div);
});

afterAll(() => {
	ReactDOM.unmountComponentAtNode(div);
	div.remove();
	div = null;
});

const startGame = jest.fn();

it('<StartGame /> renders without crashing', () => {
	const {queryByTestId} = render(<StartGame start={startGame}/>, div);

	expect(queryByTestId('start-game')).toBeTruthy();
});

it('<StartGame /> matches snapshot', () => {
	const tree = renderer.create(<StartGame start={startGame}/>, div);

	expect(tree).toMatchSnapshot();
});

describe('Search button', () => {
	it('<StartGame/> button working properly', () => {
		const {queryByTestId} = render(<StartGame start={startGame} />, div);

		const button = queryByTestId('start-game');

		expect(button.textContent).toBe('Start game!');

		fireEvent.click(button);

		expect(startGame).toHaveBeenCalledTimes(1);

		for (let i = 0; i < 5; i++) {
			fireEvent.click(button);
		}

		expect(startGame).toHaveBeenCalledTimes(6);
	});
});
