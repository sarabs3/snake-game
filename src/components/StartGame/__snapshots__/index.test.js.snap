// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StartGame /> matches snapshot 1`] = `
<div
  className="snake-app__overlay"
>
  <button
    data-testid="start-game"
    onClick={[MockFunction]}
  >
    Start game!
  </button>
</div>
`;
